hotsos/core/plugins/kernel/kernlog/calltrace.py

```diff
@@ -258,6 +258,55 @@ def __iter__(self):
             yield deadlocks
 
 
+class FanotifyDeadlockType(TraceTypeBase):
+
+    def __init__(self):
+        self._search_def = None
+        self.fanotify_hangs = []
+
+    @property
+    def name(self):
+        return 'calltrace-fanotify'
+
+    @property
+    def searchdef(self):
+        if self._search_def:
+            return self._search_def
+
+        start = SearchDef(r'.+ blocked for more than (\d+) seconds.')
+        body = SearchDef(r'(\S+) schedule(\S+) fanotify_handle_event(\S+)'
+                         r' fsnotify(\S+) security_file_open(\S+)'
```

Comment:
`fanotify_handle_event` can be reached with another call path, i.e.:
```
 #0 [ffffb57ac3147b30] __schedule at ffffffff9cda0731
 #1 [ffffb57ac3147bc0] schedule at ffffffff9cda0cc5
 #2 [ffffb57ac3147bd0] fanotify_handle_event at ffffffff9c790008
 #3 [ffffb57ac3147c38] fsnotify at ffffffff9c78c213
 #4 [ffffb57ac3147cd0] do_dentry_open at ffffffff9c73ad0e
 #5 [ffffb57ac3147d00] path_openat at ffffffff9c74f72e
 #6 [ffffb57ac3147dd8] do_filp_open at ffffffff9c751a43
 #7 [ffffb57ac3147ee0] do_sys_open at ffffffff9c73c5e4
 #8 [ffffb57ac3147f38] do_syscall_64 at ffffffff9c40430b
 #9 [ffffb57ac3147f50] entry_SYSCALL_64_after_hwframe at ffffffff9ce000ad
 ```
 which won't fit the pattern above. IMO we can omit `security_file_open(\S+) vfs_open(\S+) do_filp_open(\S+)` part.